lib/trace-agent.js

```diff
@@ -247,25 +247,22 @@ TraceAgent.prototype.parseContextFromHeader = function(str) {
 };
 
 /**
- * Adds the provided trace context to the provided http headers
- * so it can follow the associated request through other
- * Google services.
+ * Generates a trace context header value that can be used
+ * to follow the associated request through other Google services.
  *
  * @param {SpanData} spanData The span to be added to headers.
- * @param {Object} headers The http headers associated with the
- *   current request.
  * @param {boolean} traced Whether this request was traced by the agent.
  */
-TraceAgent.prototype.addContextToHeaders = function(spanData, headers, traced) {
+TraceAgent.prototype.generateTraceContext = function(spanData, traced) {
   if (spanData === SpanData.nullSpan) {
-    return;
+    return '';
```

Comment:
Actually, I'm not sure this is a good idea. The nullspan is a valid input to this function but it will cause a crash. Additionally, it will be very easy to introduce a bug that will only manifest on sampled requests which will be less likely to be caught in unit tests. I would prefer to leave this in and comment specifying that the empty header should be checked before writing to headers. WDYT?